sqlite/test/general/datetime.test.js

```diff
@@ -0,0 +1,14 @@
+const cds = require('../../../test/cds.js')
+
+cds.test(__dirname, 'testModel.cds')
+
+describe('datetime handling', () => {
+  test('datetime elements as key', async () => {
+    let res
+    const payload = { dt: '2020-12-31T01:02:03Z', int: 4711}
+    res = await cds.db.run(INSERT.into('DateTimeEntity').entries(payload))
+    res = await cds.db.run(SELECT.one.from('DateTimeEntity').where({dt: payload.dt})) // this returns undefined
```

Comment:
```suggestion
    res = await cds.db.run(SELECT.one.from('DateTimeEntity').where({dt: payload.dt})) // this returns undefined
    res = await cds.db.run(SELECT.one.from('DateTimeEntity').where({dt: new Date(payload.dt)}))
```

This should functions the same as the next line. I don't think it is a valid solution, but wanted to make sure this scenario is also included. There is also a test currently skipped in `cds` which encountered the same issue.

Reply:
update the test and comment

Reply:
The `toISOString` should not be required as `cqn2sql` should call it on any `Date` object. 

Also it should be possible to compare a `DateTime` with a `Timestamp` and find a match. As typed databased will do an implicit cast between the two types and match accordingly.

```SQL
TO_SECONDDATE('2020-12-31T01:02:03Z')     = TO_TIMESTAMP('2020-12-31T01:02:03.000Z') --> TRUE
TO_SECONDDATE('2020-12-31T01:02:03.123Z') = TO_TIMESTAMP('2020-12-31T01:02:03.000Z') --> TRUE
TO_SECONDDATE('2020-12-31T01:02:03Z')     = TO_TIMESTAMP('2020-12-31T01:02:03.123Z') --> FALSE
TO_TIMESTAMP('2020-12-31T01:02:03.123Z')  = TO_SECONDDATE('2020-12-31T01:02:03Z')    --> FALSE
```

HANA seems to cast to `Timestamp` when comparing `DateTime` with `Timestamp`. As the precision of the `Timestamp` can result in a mismatch.

Reply:
The failing one is exactly one of the cases [documented](https://pages.github.tools.sap/cap/docs/guides/databases-sqlite#comparing-datetimes-timestamps) to fail:

For example, all these work:

```js
SELECT.from(Foo).where `someTimestamp = anotherTimestamp`
SELECT.from(Foo).where `someTimestamp = someDateTime`
SELECT.from(Foo).where `someTimestamp = ${new Date}`
SELECT.from(Foo).where `someTimestamp = ${req.timestamp}`
SELECT.from(Foo).where `someTimestamp = ${'2022-11-11T11:11:11.123Z'}`
```

While these would fail, because the input values don't comply to the rules:

```js
SELECT.from(Foo).where `createdAt = ${'2022-11-11T11:11:11+02:00'}` // non-zulu time zone
SELECT.from(Foo).where `createdAt = ${'2022-11-11T11:11:11Z'}` // missing 3-digit fractions
```

This is because we never can reliable infer the types of input to where clause expressions → it's strongly recommended to use always use `Date` objects there, e.g.: 

```js
SELECT.from(Foo).where { createdAt: new Date('2022-11-11T11:11:11Z') }`
```